haystack_experimental/components/agents/agent.py

```diff
@@ -262,6 +261,9 @@ def run(
         :param kwargs: Additional keyword arguments matching the defined input types
         :return: Dictionary containing messages and outputs matching the defined output types
         """
+        if not self._is_warmed_up:
+            raise RuntimeError("The component Agent wasn't warmed up. Run 'warm_up()' before calling 'run()'.")
```

Comment:
@anakin87 what's your take on this? It's not technically needed since we eventually call self._pipeline.run() which does call `warm_up` internally. I guess good to have in case we ever decide to not use `Pipeline` to run Agent?

Reply:
It seems consistent with other components in our codebase, so I'm OK with this approach.

Reply:
After talking with @julian-risch for a similar topic about SuperComponents that it doesn't make sense to have the check since it technically isn't needed since pipeline.run will warm up the components. 

So removing for now, but could come back if we go with my proposal [here](https://github.com/deepset-ai/haystack-experimental/pull/245)